### PR TITLE
Lawyer and Notrary authorization and requirements confirmation

### PR DESCRIPTION
**Pull request by orelbn**
*Issue #:* /bcgov/entity#16839

*Description of changes:*
- Updates store and state-model to accommodate the new state
- Adds `ConfirmRequirements` and `Authorization` component to `QsReviewConfirm` including their validations
- Updates and adds tests

Note for reviewers: after logging in with BCREG0018 use this link to gain access.
https://bcregistry-assets-dev--pr-1478-ls8eb2z0.web.app/dashboard/?accountid=3144

*Screenshots:*

Initial component:
image

Updating legal name:

image

Validations:

image

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1478-ls8eb2z0.web.app

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1478-ls8eb2z0.web.app

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1478-ls8eb2z0.web.app

File: ppr-ui/src/store/state/state-model.ts
Comment (cameron-eyds) [inline ppr-ui/src/store/state/state-model.ts]: I'm not sure we need to wrap these component properties in `qsReviewConfirm`, unless there is some good value to do so?

File: ppr-ui/src/views/userAccess/QsReviewConfirm.vue
Comment (cameron-eyds) [inline ppr-ui/src/views/userAccess/QsReviewConfirm.vue:30]: Looks great, nice work on the helper classes. Much preferred to a local class 👍

File: ppr-ui/src/views/userAccess/QsReviewConfirm.vue
Comment (cameron-eyds) [inline ppr-ui/src/views/userAccess/QsReviewConfirm.vue:87]: Look at you go 👍

File: ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue]: Very nice component, good job. 

Super minor but Component names are PascalCase.

File: ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue]: Not a big deal but the only thing separating this component from being completely reusable for any list formatted like this, would be these bindings. 

You could* pass this in as a prop and you'd pretty purely atomic, your call though!

**Review by cameron-eyds — APPROVED** [review]
Great stuff! 
Just a few minor comments/suggestions.

**Review by doug-lovett — APPROVED** [review]
LGTM.

File: ppr-ui/src/store/state/state-model.ts
Comment (orelbn) [reply ppr-ui/src/store/state/state-model.ts]: Agreed ill change that

File: ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue
Comment (orelbn) [reply ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue]: Will change it to a prop.

File: ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue
Comment (cameron-eyds) [inline ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue:45]: Lean and mean 👍

File: ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue
Comment (orelbn) [reply ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue]: I changed it to a prop but I still kept the prop and component name the same `requirements`/`ListRequirements`, this can defiantly be made more abstract, but it just to hint to future developers, that they might need to make some minor changes before it can be reused, because I haven't tested or made sure that it is reusable out the box.

**Comment by cameron-eyds** [comment]
Great tidy up, thank you.
Good to merge when ready!

File: ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue
Comment (cameron-eyds) [reply ppr-ui/src/components/userAccess/ReviewConfirm/ListRequirements.vue]: Absolutely sounds good. 
Such is always the case, don't need to make everything full proof / future proof. 

That can always be done incrementally and iteratively.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1478-ls8eb2z0.web.app

File: ppr-ui/src/store/state/state-model.ts
Comment (orelbn) [reply ppr-ui/src/store/state/state-model.ts]: I set localState, and use setters to deal with modifying computed primitive value. 
As we discussed in the future if we take the time to learn and use `state` instead of `getters` in store we can probably remove a lot of the extra code, and have a better dev tools experience.

Comment (cameron-eyds) [reply ppr-ui/src/store/state/state-model.ts]: You bet, looks fine to me 👍

**Pull request merged**
